File: 1701. Average Waiting Time.go
func averageWaitingTime(customers [][]int) float64 {
	totalWait, currentTime := 0, 0
	for _, c := range customers {
		arrival, time := c[0], c[1]
		if arrival < currentTime {
			totalWait += currentTime - arrival
		}
		totalWait += time
		currentTime = max(currentTime, arrival) + time
	}
	return float64(totalWait) / float64(len(customers))
}

File: java/1701. Average Waiting Time.java
class Solution {
    public double averageWaitingTime(int[][] customers) {
        double totalWait = 0;
        int previousFinish = customers[0][0];
        for (int[] cus : customers) {
            if (previousFinish <= cus[0]) {
                previousFinish = cus[0] + cus[1];
                totalWait += cus[1];
            } else {
                totalWait += previousFinish - cus[0] + cus[1];
                previousFinish += cus[1];
            }
        }
        return totalWait / customers.length;
    }
}
